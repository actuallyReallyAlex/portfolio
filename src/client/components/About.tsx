import * as React from "react";

export interface AboutProps {}

const About: React.FunctionComponent<AboutProps> = () => {
  return (
    <div>
      <h2>About</h2>
      <img
        alt="Alex Lee profile image"
        src="https://alexlee.dev/static/3380014c62deb5ae093d75d47cb23b95/0e329/avatar-2.jpg"
      />
      <p>
        Hi. I'm Alex, a Fullstack Application Developer in San Diego. I deliver
        solutions to complex problems. I create purposeful and functional
        applications that go beyond expectations. I'm currently developing the
        future of the online banking industry with U.S. Bank. Outside of work, I
        love surfing, camping, and spending time with my dog Copper.
      </p>
    </div>
  );
};

export default About;
