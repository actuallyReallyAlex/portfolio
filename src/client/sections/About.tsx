import * as React from "react";
import { Box, Button, Flex, Heading, Image, Text } from "rebass";

export interface AboutProps {}

const About: React.FunctionComponent<AboutProps> = () => {
  return (
    <div id="about">
      <Flex alignItems="center" flexDirection="column" sx={{ margin: "100px 15%" }}>
        <Heading as="h2" sx={{ textTransform: "uppercase" }}>
          About
        </Heading>
        <Image
          alt="Alex Lee profile image"
          sx={{
            borderRadius: "50%",
            margin: "10px auto",
            objectFit: "cover",
            objectPosition: "center center",
          }}
          height={250}
          src="/assets/avatar-optimized.jpg"
          width={250}
        />
        <Box maxWidth="75%" sx={{ textAlign: "center" }}>
          <Text fontWeight="bold" margin="10px auto">
            Hi. I'm Alex, a Fullstack Application Developer in San Diego. I
            deliver solutions to complex problems.
          </Text>
          <Text margin="10px auto">
            I create purposeful and functional applications that go beyond
            expectations. I'm currently developing the future of the online
            banking industry with U.S. Bank. Outside of work, I love surfing,
            camping, and spending time with my dog Copper.
          </Text>
          <a
            href="/assets/alex-lee-resume.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            <Button
              sx={{
                marginTop: "10px",
                transition: "0.25s ease-in-out",
                ":hover": {
                  backgroundColor: "rgba(0, 119, 204, 0.8)",
                  cursor: "pointer",
                },
              }}
              variant="primary"
            >
              <i className="far fa-file-user" style={{ marginRight: "10px" }} />
              Résumé
            </Button>
          </a>
        </Box>
      </Flex>
    </div>
  );
};

export default About;
