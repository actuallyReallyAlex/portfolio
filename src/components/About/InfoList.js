import React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading, Paragraph, Text } from 'grommet'
import { about } from '../../config.json'

const InfoList = ({ size }) => {
  return (
    <Box
      align="start"
      direction={size === 'small' ? 'column' : 'row'}
      fill="horizontal"
      gap={size === 'small' ? 'medium' : 'xlarge'}
      justify="center"
      margin={{ vertical: 'large' }}
    >
      <Box>
        <Heading level="3">Profile</Heading>
        <Paragraph>{about.profile}</Paragraph>
      </Box>
      <Box>
        <Heading className="uppercase" level="3" margin="none">
          Full Name:
        </Heading>
        <Text margin={{ bottom: 'small' }}>{about.fullName}</Text>
        <Heading className="uppercase" level="3" margin="none">
          Job:
        </Heading>
        <Text margin={{ bottom: 'small' }}>{about.job}</Text>
        <Heading className="uppercase" level="3" margin="none">
          Email:
        </Heading>
        <Text>{about.email}</Text>
      </Box>
    </Box>
  )
}

InfoList.propTypes = {
  size: PropTypes.string.isRequired
}

export default InfoList
