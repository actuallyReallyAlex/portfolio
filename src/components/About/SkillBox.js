import React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading } from 'grommet'

const SkillBox = ({ color, title }) => (
  <Box align="center" direction="row" gap="small" justify="center">
    <Box background={color} height="10px" width="10px" />
    <Heading level="4" margin="none">
      {title}
    </Heading>
  </Box>
)

SkillBox.propTypes = {
  color: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}

export default SkillBox
