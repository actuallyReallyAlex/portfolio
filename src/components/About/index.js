import React from 'react'
import { Box, Paragraph, Heading, Text } from 'grommet'
import SectionHeading from '../SectionHeading'
import { about } from '../../config.json'

const About = () => {
  return (
    <Box
      align="center"
      fill="horizontal"
      id="About"
      justify="center"
      pad="large"
    >
      <SectionHeading heading="About" subHeading="Let me introduce myself." />
      <Box
        align="center"
        direction="row"
        fill="horizontal"
        gap="xlarge"
        justify="center"
        margin={{ vertical: 'large' }}
      >
        <Box
          background={`url(https://avatars2.githubusercontent.com/u/41493792?s=400&u=d55d8a934c12a5a11c0618657aa60e2efb5e89d4&v=4)`}
          height="192px"
          width="192px"
        />
        <Paragraph margin="none" size="medium">
          I'm a passionate Web Developer. Since I was 11, I've been fascinated
          with computers and design. I started writing in Visual Basic. Now I'm
          proud to say that I've grown in my development and design skillsets,
          making me a valuable asset to any team.
        </Paragraph>
      </Box>
      <Box
        align="start"
        direction="row"
        fill="horizontal"
        gap="xlarge"
        justify="center"
        margin={{ vertical: 'large' }}
      >
        <Box>
          <Heading level="3">Profile</Heading>
          <Paragraph>{about.profile}</Paragraph>
        </Box>
        <Box>
          <Heading className="uppercase" level="3" margin="none">
            Full Name:
          </Heading>
          <Text margin={{ bottom: 'small' }}>{about.fullName}</Text>
          <Heading className="uppercase" level="3" margin="none">
            Job:
          </Heading>
          <Text margin={{ bottom: 'small' }}>{about.job}</Text>
          <Heading className="uppercase" level="3" margin="none">
            Email:
          </Heading>
          <Text>{about.email}</Text>
        </Box>
      </Box>
      <Box
        align="center"
        fill="horizontal"
        justify="start"
        margin={{ vertical: 'large' }}
      >
        <Box gap="medium">
          <Heading level="3" margin="none">
            Skills
          </Heading>
          <Paragraph>{about.skillsCopy}</Paragraph>
        </Box>
      </Box>
      <Box
        align="start"
        direction="row"
        fill="horizontal"
        gap="xlarge"
        justify="center"
        margin={{ vertical: 'large' }}
      >
        {about.skills.map(({ sectionTitle, languages }, i) => (
          <Box align="start" key={i} justify="start">
            <Box align="center" fill="horizontal">
              <Heading level="3">{sectionTitle}</Heading>
            </Box>
            {languages.map(({ color, title }, i) => (
              <Box
                align="center"
                direction="row"
                gap="small"
                key={i}
                justify="center"
              >
                <Box background={color} height="10px" width="10px" />
                <Heading level="4" margin="none">
                  {title}
                </Heading>
              </Box>
            ))}
          </Box>
        ))}
      </Box>
    </Box>
  )
}

export default About
