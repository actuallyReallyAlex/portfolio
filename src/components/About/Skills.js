import React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading } from 'grommet'
import { about } from '../../config.json'
import SkillBox from './SkillBox.js'

const Skills = ({ size }) => (
  <Box
    align={size === 'small' ? 'center' : 'start'}
    direction={size === 'small' ? 'column' : 'row'}
    fill="horizontal"
    gap={size === 'small' ? 'large' : 'xlarge'}
    justify="center"
    margin={{ vertical: 'large' }}
  >
    {about.skills.map(({ sectionTitle, languages }, i) => (
      <Box align="start" key={i} justify="start">
        <Box align="center" margin={{ bottom: 'small' }}>
          <Heading level="3" margin={{ bottom: 'xsmall' }}>
            {sectionTitle}
          </Heading>
          <Box
            background="#eb1736"
            fill="horizontal"
            height="3px"
            round="small"
          />
        </Box>
        {languages.map(({ color, title }, i) => (
          <SkillBox color={color} key={i} title={title} />
        ))}
      </Box>
    ))}
  </Box>
)

Skills.propTypes = {
  size: PropTypes.string.isRequired
}

export default Skills
