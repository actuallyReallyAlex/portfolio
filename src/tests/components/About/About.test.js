import React from 'react'
import About from '../../../components/About'
import { shallow } from 'enzyme'

const defaultProps = {}

describe('<About />', () => {
  test('Should render the <About />.', () => {
    const component = shallow(<About {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
