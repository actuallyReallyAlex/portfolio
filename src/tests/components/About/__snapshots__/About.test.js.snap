// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> Should render the <About />. 1`] = `ShallowWrapper {}`;
